### DISCUSSION TITLE
Unique entry type for one-off entries in structures

### DISCUSSION THREAD

**Discussion by foxtrotcz** [craftcms/cms#13705] [Ideas]
Hello,
entry types are very good for multiple entries where their content is different but they have the same entry type.

But sometimes you can have complex structure of entries that consists of unique entries with different entry type.
Lets say the the structure consists of 10 entries forming a tree. 5 entries are each unique entry type and 5 entries are the same entry type.
Right now I have to create 6 entry types, but 5 of them will be used only once. It will just clutter the entry type menu with types I will not use repeatedly.
If there was some kind of "unique entry type" that would enable customizing entries on one-off basis I could just create 5 unique entries, each consisting of different fields. Each basically different entry type, but without cluttering the control panel with 5 entry types. And for the 5 repeating entries I would create regular entry type.

Like I said, could be good for for creating structures where entry types doesnt repeat themselves and each entry is unique. 
Thanks.

**Comment by thupsi** [comment]
I think this will potentially be resolved with the introduction of the Unified Content View in Craft 5. 

It's one of the upcoming features I'm most exited about, but, unfortunately, information is sparse for the time being. From what I can tell, after watching Brandon's presentation last year, you will be able to mix singles and other section types in a custom tree view in the CP. 

I guess/hope we will have much more info about this in a few days, when the 2023 DotAll takes place.

**Comment by brandonkelly** [comment]
This seems similar to [this suggestion](https://github.com/craftcms/cms/discussions/8099#discussioncomment-3050824) for having a condition that determines when certain Matrix block types should be available. The same idea could be applied to entry types: you could create a condition that only allows the entry type to be visible/selectable for entries with certain UIDs/IDs/slugs/etc.

**Comment by foxtrotcz** [comment]
I think I didnt explain myself clear enough and maybe I shouldnt have propose perceived "solution".
I think it will be better if I just explain what made me write this proposal.

Lets say you have multiple entries (products, list of places, encyclopedia entries...). These entries have some properties same, some different and you want to model them efficiently in Craft. Since we talk about data Table field would likely suit best.

Here is diagram.
Tree structure on top and flat structure (as seen on frontend) on bottom.

![image](https://github.com/craftcms/cms/assets/10333810/d8fc2ec3-ce20-4837-90e1-a8095f4c34ec)

Example:
Entry: Product 1
Table Row Heading: E
Table Row Value: 1

As you can see, if you just made all products in flat structure with properties in regular Table field it would be nightmare to update. Every change would have to be applied to multiple products.
Thats why I modeled it into a tree by using Craft structures. On frontend each Product goes backwards to their parents and prints all inherited data.

Next I noticed that you still have to manually update all the Table Row Headings in the bottom layer if changed. That was resolved here https://github.com/craftcms/cms/pull/13231
Now it's possible to edit Table Row Headings directly in Field settings and it applies to all instances.

The last thing I was trying to solve when creating this thread was number of entry types when modeling this content.
Right now I can see 6 entry types needed.
Entry type 1: Category 1
Entry type 2: Category 2
Entry type 3: Category 3a, 3c
Entry type 4: Category 3b
Entry type 5: Category 4a, 4b
Entry type 6: Products

Products make sense, there is a lot of them. Perfect fit for making Entry Type.
But why should I create unique entry type for Category 1, Category 2 and Category 3b? They are used only once. It will just clutter Entry Types with something I wont need more than once.

And this was the reason for my post. I tried to come up with some solution so I dont have to create multiple one-off entry types.

Recently I tried again to come up with something and it's kinda solvable with Matrix field.
I can create single Entry Type containing Matrix field that has all needed configured states inside.
![image](https://github.com/craftcms/cms/assets/10333810/22c678bb-1af8-40fa-a7f9-8cc7b77f7326)

This way I use only one Entry Type and select appropriate Table for each category in content editor.

Well now when I think about it I guess I dont need to use Static Row Headings for Category 1, Category 2 and Category 3b. These entries have only one instance and dont repeat itself. In that case I could just use one Entry Type with regular Table and fill it out for each entry.
But if they did repeat the Matrix solution will still work.

So this is what I discovered until now. If there is better solution I dont see it.
Matrix field is really flexible. I just didnt think about using it for defining structure this way.

Well, if someone has better idea please share.

Bye and thanks for reading.
